cli/cluster/cluster.go

```diff
@@ -0,0 +1,244 @@
+		return nil, fmt.Errorf("No running instances found")
+	}
+
+	instanceName := runningInstancesNames[0]
+	conn, err := ConnectToInstance(instanceName, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("Connect all instances to membership")
+
+	if err := ConnectToMembership(conn, runningInstancesNames, instancesConf); err != nil {
+		return nil, fmt.Errorf("Failed to connect instances to membership: %s", err)
+	}
+
+	membershipInstances, err := GetMembershipInstancesFromConn(conn)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get membership instances: %s", err)
+	}
+
+	return membershipInstances, nil
+}
+
+func getRunningInstances(instancesConf *InstancesConf, ctx *context.Ctx) []string {
+	var runningInstancesNames []string
+	for instanceName := range *instancesConf {
+		process := running.NewInstanceProcess(ctx, instanceName)
+		if process.IsRunning() {
+			runningInstancesNames = append(runningInstancesNames, instanceName)
+		}
+	}
+
+	return runningInstancesNames
+}
+
+func ConnectToInstance(instanceName string, ctx *context.Ctx) (*connector.Conn, error) {
+	consoleSockPath := project.GetInstanceConsoleSock(ctx, instanceName)
+	conn, err := connector.Connect(consoleSockPath, connector.Opts{})
+	if err != nil {
+		return nil, fmt.Errorf("Failed to connect to Tarantool instance: %s", err)
+	}
+
+	log.Debugf("Connected to %s", consoleSockPath)
+
+	return conn, nil
+}
+
+func HealthCheckIsNeeded(conn *connector.Conn) (bool, error) {
+	majorCartridgeVersion, err := common.GetMajorCartridgeVersion(conn)
+	if err != nil {
+		return false, fmt.Errorf("Failed to get Cartridge major version: %s", err)
+	}
+
+	return majorCartridgeVersion < 2, nil
+}
+
+func ConnectToMembership(conn *connector.Conn, runningInstancesNames []string, instancesConf *InstancesConf) error {
+	// probe all instances mentioned in topology
+	var urisToProbe []string
+
+	for _, instanceName := range runningInstancesNames {
+		instanceConf, found := (*instancesConf)[instanceName]
+		if !found {
+			return fmt.Errorf("Instance %s  isn't found in instances config", instanceName)
+		}
+
+		urisToProbe = append(urisToProbe, instanceConf.URI)
+	}
+
+	if _, err := conn.Exec(connector.EvalReq(probeInstancesBody, urisToProbe)); err != nil {
+		return fmt.Errorf("Failed to probe all instances mentioned in replica sets: %s", err)
+	}
+
+	return nil
+}
+
+func GetMembershipInstancesFromConn(conn *connector.Conn) (*MembershipInstances, error) {
+	var membershipInstancesSlice []*MembershipInstance
+
+	req := connector.EvalReq(getMembershipInstancesBody).SetReadTimeout(SimpleOperationTimeout)
+	if err := conn.ExecTyped(req, &membershipInstancesSlice); err != nil {
+		return nil, fmt.Errorf("Failed to get membership members: %s", err)
+	}
+
+	membershipInstances := make(MembershipInstances)
+	for _, membershipInstance := range membershipInstancesSlice {
+		membershipInstances[membershipInstance.URI] = membershipInstance
+	}
+
+	return &membershipInstances, nil
+}
```

Comment:
Let's separate it to three files - `membership.go`, `instances.go` and `cluster.go`.